src/index.js

```diff
@@ -17,7 +17,7 @@ const { hasTypeModule } = require('./utils/has-type-module');
 // support glob graceful-fs
 fs.gracefulify(require("fs"));
 
-const SUPPORTED_EXTENSIONS = [".js", ".json", ".node", ".mjs", ".ts", ".tsx"];
+const SUPPORTED_EXTENSIONS = [".js", ".json",".jsonc", ".node", ".mjs", ".ts", ".tsx"];
```

Comment:
Does Node.js support jsonc files?

If not, then this seems to be out of scope for ncc.